### PR TITLE
feat: add django 4.2 support

### PR DESCRIPTION
**Pull request by vinitkumar**
## Description

Add Django 4.2 to the mix and ensure all tests run under it. 

## Checklist

* [x] I have opened this pull request against ``develop``
* [x] I have added or modified the tests when changing logic
* [x] I have followed [the conventional commits guidelines](https://www.conventionalcommits.org/) to add meaningful information into the changelog
* [x] I have read the [contribution guidelines ](https://github.com/django-cms/django-cms/blob/develop/CONTRIBUTING.rst) and I have joined #workgroup-pr-review on [Slack](https://www.django-cms.org/slack) to find a “pr review buddy” who is going to review my pull request.

### CODE REVIEW

**Comment by vinitkumar** [comment]
Can't get this to work on local but somehow it works here. Strange. Don't merge @fsbraun @marksweb

**Comment by marksweb** [comment]
@vinitkumar I think Fabian has started to look into 4.2 support today and there may be issues with the test runner.

The app helper package used is supposed to be pytest compatible, but last time I tried to update the test suite to use tox and pytest it wasn't successful.

**Comment by fsbraun** [comment]
@vinitkumar What type of error do you get if you run tests locally? What happens if you only run a selected test, say. `./manage.py test cms.tests.test_cache`?

**Comment by vinitkumar** [comment]
> @vinitkumar What type of error do you get if you run tests locally? What happens if you only run a selected test, say. `./manage.py test cms.tests.test_cache`?

Errors like these:


```python
The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/vinitkumar/projects/python/django-cms/venv/lib/python3.10/site-packages/Django-4.2a1-py3.10.egg/django/test/utils.py", line 461, in inner
    return func(*args, **kwargs)
  File "/Users/vinitkumar/projects/python/django-cms/cms/tests/test_apphooks.py", line 422, in test_get_sub_page_for_apphook_with_implicit_current_app
    en_title = self.create_base_structure(NS_APP_NAME, 'en', 'namespaced_app_ns')
  File "/Users/vinitkumar/projects/python/django-cms/cms/tests/test_apphooks.py", line 99, in create_base_structure
    superuser = get_user_model().objects.create_superuser('admin', 'admin@admin.com', 'admin')
  File "/Users/vinitkumar/projects/python/django-cms/venv/lib/python3.10/site-packages/Django-4.2a1-py3.10.egg/django/contrib/auth/models.py", line 172, in create_superuser
    return self._create_user(username, email, password, **extra_fields)
  File "/Users/vinitkumar/projects/python/django-cms/venv/lib/python3.10/site-packages/Django-4.2a1-py3.10.egg/django/contrib/auth/models.py", line 155, in _create_user
    user.save(using=self._db)
  File "/Users/vinitkumar/projects/python/django-cms/venv/lib/python3.10/site-packages/Django-4.2a1-py3.10.egg/django/contrib/auth/base_user.py", line 75, in save
    super().save(*args, **kwargs)
  File "/Users/vinitkumar/projects/python/django-cms/venv/lib/python3.10/site-packages/Django-4.2a1-py3.10.egg/django/db/models/base.py", line 814, in save
    self.save_base(
  File "/Users/vinitkumar/projects/python/django-cms/venv/lib/python3.10/site-packages/Django-4.2a1-py3.10.egg/django/db/models/base.py", line 877, in save_base
    updated = self._save_table(
  File "/Users/vinitkumar/projects/python/django-cms/venv/lib/python3.10/site-packages/Django-4.2a1-py3.10.egg/django/db/models/base.py", line 1020, in _save_table
    results = self._do_insert(
  File "/Users/vinitkumar/projects/python/django-cms/venv/lib/python3.10/site-packages/Django-4.2a1-py3.10.egg/django/db/models/base.py", line 1061, in _do_insert
    return manager._insert(
  File "/Users/vinitkumar/projects/python/django-cms/venv/lib/python3.10/site-packages/Django-4.2a1-py3.10.egg/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/Users/vinitkumar/projects/python/django-cms/venv/lib/python3.10/site-packages/Django-4.2a1-py3.10.egg/django/db/models/query.py", line 1805, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
  File "/Users/vinitkumar/projects/python/django-cms/venv/lib/python3.10/site-packages/Django-4.2a1-py3.10.egg/django/db/models/sql/compiler.py", line 1787, in execute_sql
    cursor.execute(sql, params)
  File "/Users/vinitkumar/projects/python/django-cms/venv/lib/python3.10/site-packages/Django-4.2a1-py3.10.egg/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(
  File "/Users/vinitkumar/projects/python/django-cms/venv/lib/python3.10/site-packages/Django-4.2a1-py3.10.egg/django/db/backends/utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/Users/vinitkumar/projects/python/django-cms/venv/lib/python3.10/site-packages/Django-4.2a1-py3.10.egg/django/db/backends/utils.py", line 83, in _execute
    self.db.validate_no_broken_transaction()
  File "/Users/vinitkumar/projects/python/django-cms/venv/lib/python3.10/site-packages/Django-4.2a1-py3.10.egg/django/db/backends/base/base.py", line 531, in validate_no_broken_transaction
    raise TransactionManagementError(
django.db.transaction.TransactionManagementError: An error occurred in the current transaction. You can't execute queries until the end of the 'atomic' block.

The above exception was the direct cause of the following exception:```

File: test_requirements/django-4.2.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+-r requirements_base.txt
+Django==4.2a1
```
Comment (fsbraun) [inline test_requirements/django-4.2.txt:3]:
```suggestion
Django==4.2
```

**Review by fsbraun — COMMENTED** [review]
Hi @vinit! I have made a change proposal to get rid of most of the errors. It seems the root cause is that the menu misses the language in tests.

**Comment by fsbraun** [comment]
@vinitkumar Please also check out https://github.com/django-cms/django-cms/pull/7517 for changes that worked for v4, especially `views.py`, `pageadmin.py`, and `menu_pool.py`. These should allow the tests to pass.

**Comment by fsbraun** [comment]
@vinitkumar Can you also take a look at the compatibility matrix in the docs (index.rst) and update it? V4 will also be Django 4.2 compatible.

**Review by fsbraun — APPROVED** [review]

**Pull request merged**